### PR TITLE
SGJ Rendering fixes & refactor pt. 1

### PR DESCRIPTION
**Pull request by sp0lsh**
Post SGJ17 fixes (texture stripes, directional light) and created sandbox game for rendering tests.

Doing in smaller chunks, to avoid merge hell and check if builds on all platforms.

### CODE REVIEW

File: PolyEngine/Core/Src/BasicMath.hpp [hunk 1]
```diff
@@ -83,14 +83,21 @@ namespace Poly {
...
 		HEAVY_ASSERTE(min <= max, "Min is larger than max when calling clamp!");
 		return val < min ? min : (val > max ? max : val);
 	}
+
 	template <typename T> inline T Lerp(const T& val1, const T& val2, float t) {
-		HEAVY_ASSERTE(t >= 0.f && t <= 1.f, "Value of t needs to be between 0 and 1.");
 		return val1 * (1.0f - t) + val2 * t;
 	}
 
+	template <typename T> inline T Smoothstep(const T& edge1, const T& edge2, float x) {
```
Comment (MuniuDev) [inline PolyEngine/Core/Src/BasicMath.hpp]: Shouldn't `x` be also of type `const T&` ? Also rename to `SmoothStep`

File: PolyEngine/Engine/Res/Shaders/blinn-phongFrag.shader [hunk 2]
```diff
@@ -35,50 +36,55 @@ in vec3 vNormal;
...
+	// 	point += texColor * diffuseLight(uPointLight[i].Base, normalize(lightPos - vVertexPos), normalWS)
+	// 		* pointLightAttenuation(uPointLight[i], vVertexPos);
+	// }
+
+	// color = point + uBaseColor;
+	// color = texColor;
+	// color = vec4(0.5*(1.0+normalWS), 1.0);
+	// color = vec4(max(dot(normalWS, uDirectionalLight.Direction.xyz), 0.0));
+
+	// color = vec4(pointNdotL);
```
Comment (MuniuDev) [inline PolyEngine/Engine/Res/Shaders/blinn-phongFrag.shader]: Let's remove commented code.

File: PolyEngine/Core/Src/BasicMath.hpp (hunk 1 above)
Comment (sp0lsh) [inline PolyEngine/Core/Src/BasicMath.hpp]: ok

File: PolyEngine/Engine/Res/Shaders/blinn-phongFrag.shader (hunk 2 above)
Comment (sp0lsh) [inline PolyEngine/Engine/Res/Shaders/blinn-phongFrag.shader]: ok, that is work in progress

**Pull request merged**